std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -477,6 +526,61 @@ public static List<GraphemeSpan> span_of_all_case_insensitive(
     return result;
   }
 
+  /**
+   * Find spans of all occurrences of a set of needles within the haystack,
+   * case-insensitively.
+   *
+   * @param haystack the string to search
+   * @param needle the substring that is searched for
+   * @param locale the locale used for case-insensitive comparisons
+   * @return a list of extended-grapheme-cluster spans at which the needle occurs in the haystack
+   */
+  public static List<GraphemeSpan> span_of_all_case_insensitive_multiple(
+          String haystack, List<String> needles, Locale locale) {
+    if (needles.isEmpty())
+      throw new IllegalArgumentException(
+              "The operation `span_of_all_case_insensitive_multiple` does not support searching for an empty term.");
+    for (String needle : needles) {
+      if (needle.isEmpty())
+        throw new IllegalArgumentException(
+                "The operation `span_of_all_case_insensitive_multiple` does not support searching for an empty term.");
+    }
+    if (haystack.isEmpty()) return List.of();
+
+    CaseFoldedString foldedHaystack = CaseFoldedString.fold(haystack, locale);
+    List<String> foldedNeedles = IntStream.range(0, needles.size())
+            .mapToObj(i -> CaseFoldedString.simpleFold(needles.get(i), locale))
+            .collect(Collectors.toList());
+    List<StringSearch> stringSearches = IntStream.range(0, foldedNeedles.size())
+            .mapToObj(i -> new StringSearch(foldedNeedles.get(i), foldedHaystack.getFoldedString()))
+            .collect(Collectors.toList());
+
+    List<GraphemeSpan> occurrences = new ArrayList<>();
+
+    int ix = 0;
+    while (ix != StringSearch.DONE) {
+      int earliestIndex = -1;
+      int earliestStart = -1;
+      for (int i = 0; i < stringSearches.size(); ++i) {
+        StringSearch stringSearch = stringSearches.get(i);
+        int start = stringSearch.following(ix);
+        if (start != StringSearch.DONE && (earliestStart == -1 || start < earliestStart)) {
+          earliestIndex = i;
+          earliestStart = start;
+        }
+      }
+      if (earliestIndex == -1) {
+        // No more matches.
+        break;
+      }
+      int matchLength = stringSearches.get(earliestIndex).getMatchLength();
+      GraphemeSpan graphemeSpan = findExtendedSpan(foldedHaystack, earliestStart, matchLength);
+      occurrences.add(graphemeSpan);
+      ix = graphemeSpan.codeunit_end;
+    }
+    return occurrences;
+  }
```

Comment:
From a DRY perspective:

```suggestion
  public static List<GraphemeSpan> span_of_all_case_insensitive_multiple(
          String haystack, List<String> needles, Locale locale) {
    try {
      CaseFoldedString foldedHaystack = CaseFoldedString.fold(haystack, locale);
      List<String> foldedNeedles = needles.stream()
              .map(needle -> CaseFoldedString.simpleFold(needle, locale))
              .collect(Collectors.toList());
      var foldedSpans = span_of_all(foldedHaystack.foldedString, foldedNeedles);

      List<GraphemeSpan> occurrences =
              foldedSpans.stream()
                      .map(span -> findExtendedSpan(foldedHaystack, span.start, span.length))
                      .collect(Collectors.toList());
      return occurrences;
    } catch (IllegalArgumentException e) {
      throw new IllegalArgumentException(
              "The operation `span_of_all_case_insensitive_multiple` does not support searching for an empty term.");
    }
  }
```

No guarantee that works as wrote in a text editor!